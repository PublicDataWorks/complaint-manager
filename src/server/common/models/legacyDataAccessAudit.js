/* 
****** This model should stay in the code base even after the
  newAuditFeature is removed 
 */

"use strict";

module.exports = (sequelize, DataTypes) => {
  const LegacyDataAccessAudit = sequelize.define(
    "legacy_data_access_audit",
    {
      id: {
        allowNull: false,
        autoIncrement: true,
        primaryKey: true,
        type: DataTypes.INTEGER
      },
      auditSubject: {
        field: "audit_subject",
        allowNull: false,
        type: DataTypes.STRING
      },
      auditDetails: {
        field: "audit_details",
        type: DataTypes.JSONB
      }
    },
    {
      tableName: "legacy_data_access_audits",
      timestamps: false
    }
  );

  LegacyDataAccessAudit.associate = models => {
    LegacyDataAccessAudit.belongsTo(models.audit, {
      as: "legacyDataAccessAudit",
      foreignKey: { name: "auditId", field: "audit_id" }
    });
  };

  return LegacyDataAccessAudit;
};
