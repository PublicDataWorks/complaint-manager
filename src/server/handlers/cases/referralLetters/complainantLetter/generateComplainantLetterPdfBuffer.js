import { OFFICER_COMPLAINANT_TITLE } from "../../../../../sharedUtilities/constants";
import fs from "fs";
import Handlebars from "handlebars";
import generatePdfBuffer from "../sharedLetterUtilities/generatePdfBuffer";
import { getPersonType } from "../../../../policeDataManager/models/modelUtilities/getPersonType";

require("../../../../handlebarHelpers");

const generateComplainantLetterPdfBuffer = async (
  existingCase,
  complainant
) => {
  const letterHtml = await generateComplainantLetterHtml(
    existingCase,
    complainant
  );
  return await generatePdfBuffer(letterHtml);
};

export const generateComplainantLetterHtml = async (
  existingCase,
  complainant
) => {
  const pdfData = getComplainantLetterPdfData(existingCase, complainant);

  const rawTemplate = fs.readFileSync(
    `${process.env.REACT_APP_INSTANCE_FILES_DIR}/complainantLetterPdf.tpl`
  );
  const compiledTemplate = Handlebars.compile(rawTemplate.toString());
  return compiledTemplate(pdfData);
};

const getComplainantLetterPdfData = (existingCase, complainant) => {
  const currentDate = Date.now();

  let revisedTitle;
  if (complainant.civilianTitle && complainant.civilianTitle.name !== "N/A") {
    revisedTitle = complainant.civilianTitle.name;
  } else if (
    !complainant.civilianTitle ||
    (complainant.civilianTitle.name && complainant.civilianTitle.name === "N/A")
  ) {
    revisedTitle = "";
  } else {
    revisedTitle = OFFICER_COMPLAINANT_TITLE;
  }

  return {
    caseReference: existingCase.caseReference,
    recipientFirstName: complainant.firstName,
    recipientLastName: complainant.lastName,
    currentDate: currentDate,
    complainantAddress: complainant.address ? complainant.address : null,
    complainantEmail: complainant.email ? complainant.email : null,
    firstContactDate: existingCase.firstContactDate,
    title: revisedTitle,
    complainantPersonType: getPersonType(complainant)
  };
};

export default generateComplainantLetterPdfBuffer;
