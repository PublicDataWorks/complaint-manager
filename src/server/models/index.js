const fs = require("fs");
const path = require("path");
const Sequelize = require("sequelize");
const basename = path.basename(__filename);
let env = process.env.NODE_ENV || "development";
if (env === "test") {
  env += process.env.JEST_WORKER_ID;
}
const config = require(__dirname + "/../config/sequelize_config.js")[env];
const db = {};
const dataChangeAuditHooks = require("../sequelizeHooks/dataChangeAuditHooks");
const caseStatusHooks = require("../sequelizeHooks/caseStatusHooks");

let sequelize;
if (config.use_env_variable) {
  sequelize = new Sequelize(process.env[config.use_env_variable], config);
} else {
  sequelize = new Sequelize(
    config.database,
    config.username,
    config.password,
    config
  );
}

//TODO Refactor to pass in only the class references we need.
dataChangeAuditHooks.init(sequelize, Sequelize.Model);
caseStatusHooks.init(sequelize, Sequelize.Model);

fs.readdirSync(__dirname)
  .filter(file => {
    return (
      file.indexOf(".") !== 0 && file !== basename && file.slice(-3) === ".js"
    );
  })
  .forEach(file => {
    var model = sequelize["import"](path.join(__dirname, file));
    db[model.name] = model;
  });

Object.keys(db).forEach(modelName => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
