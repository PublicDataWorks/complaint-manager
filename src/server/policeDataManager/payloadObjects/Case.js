import Boom from "boom";
import { BAD_REQUEST_ERRORS } from "../../../sharedUtilities/errorMessageConstants";
import determineNextCaseStatus from "../../handlers/cases/helpers/determineNextCaseStatus";

export default class Case {
  constructor(caseModel) {
    // TODO maybe allow the constructor to get the model by id from DB (if needed)
    this._model = caseModel;
    this._status = "Initial";
  }

  get model() {
    return this._model;
  }

  get id() {
    return this._model.id;
  }

  get primaryComplainant() {
    return this._model.primaryComplainant;
  }

  get complaintType() {
    return this._model.complaintType;
  }

  get status() {
    return this._status;
  }

  setStatus = async status => {
    const nextStatus = await determineNextCaseStatus(this.status);
    if (status === nextStatus.name) {
      this._model.currentStatusId = nextStatus.id;
      this._status = nextStatus.name;
    } else if (status !== this.status) {
      throw Boom.badRequest(BAD_REQUEST_ERRORS.INVALID_CASE_STATUS);
    }
  };

  get nextStatus() {
    return determineNextCaseStatus(this.status);
  }

  get year() {
    return this._model.year;
  }

  get caseNumber() {
    return this._model.caseNumber;
  }

  get caseReferencePrefix() {
    return this._model.caseReferencePrefix;
  }

  get caseReference() {
    return this._model.caseReference;
  }

  get firstContactDate() {
    return this._model.firstContactDate;
  }

  get incidentDate() {
    return this._model.incidentDate;
  }

  get intakeSourceId() {
    return this._model.intakeSourceId;
  }

  get districtId() {
    return this._model.districtId;
  }

  get incidentTime() {
    return this._model.incidentTime;
  }

  get incidentTimezone() {
    return this._model.incidentTimezone;
  }

  get narrativeSummary() {
    return this._model.narrativeSummary;
  }

  get narrativeDetails() {
    return this._model.narrativeDetails;
  }

  get pibCaseNumber() {
    return this._model.pibCaseNumber;
  }

  get createdBy() {
    return this._model.createdBy;
  }

  get assignedTo() {
    return this._model.assignedTo;
  }
}
