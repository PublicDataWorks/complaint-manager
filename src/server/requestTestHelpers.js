import jwt from "jsonwebtoken";
import fs from "fs";
import path from "path";

const config = require("./config/config")[process.env.NODE_ENV];

function buildTokenWithPermissions(permissions, nickname) {
  const privateKeyPath = path.join(__dirname, "config", "test", "private.pem");
  const cert = fs.readFileSync(privateKeyPath);

  const payload = {
    foo: "bar",
    scope: `${config.authentication.scope} ${permissions}`
  };
  payload[`${config.authentication.nicknameKey}`] = nickname;

  const options = {
    audience: config.authentication.audience,
    issuer: config.authentication.issuer,
    algorithm: config.authentication.algorithm
  };

  return jwt.sign(payload, cert, options);
}

module.exports = buildTokenWithPermissions;
