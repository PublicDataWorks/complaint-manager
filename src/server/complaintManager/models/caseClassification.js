import models from "./index";

module.exports = (sequelize, DataTypes) => {
  const CaseClassification = sequelize.define(
    "case_classification",
    {
      caseId: {
        allowNull: false,
        type: DataTypes.INTEGER,
        field: "case_id",
        references: {
          model: models.cases,
          key: "id"
        }
      },
      classificationId: {
        allowNull: false,
        type: DataTypes.INTEGER,
        field: "classification_id",
        references: {
          model: models.classification,
          key: "id"
        }
      },
      createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        field: "created_at"
      },
      updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        field: "updated_at"
      },
      deletedAt: {
        type: DataTypes.DATE,
        field: "deleted_at"
      }
    },
    { tableName: "case_classifications", paranoid: true }
  );

  CaseClassification.associate = models => {
    CaseClassification.belongsTo(models.classification, {
      as: "classification",
      foreignKey: {
        name: "classificationId",
        field: "classification_id",
        allowNull: false
      }
    });
  };

  CaseClassification.prototype.getCaseId = async function(transaction) {
    return this.caseId;
  };

  CaseClassification.prototype.modelDescription = async function(transaction) {
    const classification = await sequelize
      .model("classification")
      .findByPk(this.classificationId, {
        transaction
      });
    return [
      { "Classification Name": classification.name },
      { Message: classification.message }
    ];
  };

  CaseClassification.auditDataChange();
  return CaseClassification;
};
