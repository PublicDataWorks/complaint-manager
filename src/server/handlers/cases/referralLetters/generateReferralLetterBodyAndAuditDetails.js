import models from "../../../models";
import fs from "fs";
import Handlebars from "handlebars";
import { ASCENDING } from "../../../../sharedUtilities/constants";
import getQueryAuditAccessDetails from "../../audits/getQueryAuditAccessDetails";

export const getReferralLetterCaseDataAndAuditDetails = async (
  caseId,
  transaction
) => {
  const queryOptions = {
    attributes: [
      "id",
      "incidentDate",
      "incidentTime",
      "narrativeDetails",
      "firstContactDate",
      "complaintType",
      "year",
      "caseNumber",
      "pibCaseNumber",
      "caseReference"
    ],
    order: [
      [
        { model: models.case_officer, as: "complainantOfficers" },
        "createdAt",
        ASCENDING
      ],
      [
        { model: models.civilian, as: "complainantCivilians" },
        "createdAt",
        ASCENDING
      ],
      [
        { model: models.civilian, as: "witnessCivilians" },
        "createdAt",
        ASCENDING
      ],
      [
        { model: models.case_officer, as: "witnessOfficers" },
        "createdAt",
        ASCENDING
      ]
    ],
    include: [
      {
        model: models.referral_letter,
        as: "referralLetter",
        include: [
          {
            model: models.referral_letter_iapro_correction,
            as: "referralLetterIaproCorrections"
          }
        ]
      },
      { model: models.classification },
      {
        model: models.address,
        as: "incidentLocation"
      },
      {
        model: models.civilian,
        as: "complainantCivilians",
        include: [
          { model: models.address },
          { model: models.race_ethnicity, as: "raceEthnicity" },
          { model: models.gender_identity, as: "genderIdentity" }
        ]
      },
      {
        model: models.civilian,
        as: "witnessCivilians"
      },
      {
        model: models.case_officer,
        as: "complainantOfficers"
      },
      {
        model: models.case_officer,
        as: "accusedOfficers",
        separate: true,
        include: [
          {
            model: models.officer_allegation,
            as: "allegations",
            include: [{ model: models.allegation }]
          },
          {
            model: models.letter_officer,
            as: "letterOfficer",
            include: [
              {
                model: models.referral_letter_officer_history_note,
                as: "referralLetterOfficerHistoryNotes",
                separate: true
              },
              {
                model: models.referral_letter_officer_recommended_action,
                as: "referralLetterOfficerRecommendedActions",
                separate: true,
                include: [
                  {
                    model: models.recommended_action,
                    as: "recommendedAction"
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        model: models.case_officer,
        as: "witnessOfficers"
      }
    ],
    transaction
  };
  const caseData = await models.cases.findByPk(caseId, queryOptions);

  const auditDetails = getQueryAuditAccessDetails(
    queryOptions,
    models.cases.name
  );

  return { caseData: caseData, auditDetails: auditDetails };
};

const referralLetterBodyPath =
  "src/server/handlers/cases/referralLetters/getReferralLetterPreview/letterBody.tpl";

export const generateReferralLetterBodyAndAuditDetails = async (
  caseId,
  transaction
) => {
  let caseData;

  const caseDataAndAuditDetails = await getReferralLetterCaseDataAndAuditDetails(
    caseId,
    transaction
  );

  caseData = caseDataAndAuditDetails.caseData.toJSON();
  caseData.accusedOfficers.sort((officerA, officerB) => {
    return officerA.createdAt > officerB.createdAt;
  });

  const rawTemplate = fs.readFileSync(referralLetterBodyPath);
  const compiledTemplate = Handlebars.compile(rawTemplate.toString());
  return {
    referralLetterBody: compiledTemplate(caseData),
    auditDetails: caseDataAndAuditDetails.auditDetails
  };
};
