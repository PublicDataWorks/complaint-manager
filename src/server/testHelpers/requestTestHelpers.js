import jwt from "jsonwebtoken";
import fs from "fs";
import path from "path";
import models from "../policeDataManager/models/index";
import winston from "winston";

const config = require("../config/config")[process.env.NODE_ENV];

export const suppressWinstonLogs = test => async () => {
  winston.configure({
    transports: [
      new winston.transports.Console({
        json: config.winston.json,
        colorize: true,
        silent: true
      })
    ],
    level: config.winston.logLevel,
    colorize: true
  });

  try {
    await test();
  } catch (err) {
    throw err;
  } finally {
    winston.configure({
      transports: [
        new winston.transports.Console({
          json: config.winston.json,
          colorize: true
        })
      ],
      level: config.winston.logLevel,
      colorize: true
    });
  }
};

export const buildTokenWithPermissions = (permissions, nickname) => {
  const privateKeyPath = path.join(
    __dirname,
    "../config",
    "test",
    "private.pem"
  );
  const cert = fs.readFileSync(privateKeyPath);

  const payload = {
    foo: "bar",
    scope: `${config.authentication.scope} ${permissions}`
  };
  payload[`${config.authentication.nicknameKey}`] = nickname;

  const options = {
    audience: config.authentication.audience,
    issuer: config.authentication.issuer,
    algorithm: config.authentication.algorithm
  };

  return jwt.sign(payload, cert, options);
};

export const expectResponse = async (
  responsePromise,
  statusCode,
  responseBodyContents = null
) => {
  return await responsePromise.then(
    response => {
      expect(response.statusCode).toEqual(statusCode);
      responseBodyContents &&
        expect(response.body).toEqual(responseBodyContents);
      return response;
    },
    error => {
      expect(error.response.statusCode).toEqual(statusCode);
      responseBodyContents &&
        expect(error.response.body).toEqual(responseBodyContents);
      return error;
    }
  );
};

export const cleanupDatabase = async () => {
  const truncationQuery =
    "TRUNCATE referral_letter_officer_recommended_actions CASCADE;" +
    "TRUNCATE referral_letter_officer_history_notes CASCADE;" +
    "TRUNCATE letter_officers CASCADE;" +
    "TRUNCATE referral_letters CASCADE;" +
    "TRUNCATE recommended_actions CASCADE;" +
    "TRUNCATE addresses CASCADE;" +
    "TRUNCATE cases_officers CASCADE;" +
    "TRUNCATE officers_allegations CASCADE;" +
    "TRUNCATE officers CASCADE;" +
    "TRUNCATE allegations CASCADE;" +
    "TRUNCATE classifications CASCADE;" +
    "TRUNCATE civilians CASCADE;" +
    "TRUNCATE attachments CASCADE;" +
    "TRUNCATE case_notes CASCADE;" +
    "TRUNCATE action_audits CASCADE;" +
    "TRUNCATE legacy_data_change_audits CASCADE;" +
    "TRUNCATE intake_sources CASCADE;" +
    "TRUNCATE how_did_you_hear_about_us_sources CASCADE;" +
    "TRUNCATE race_ethnicities CASCADE;" +
    "TRUNCATE officer_history_options CASCADE;" +
    "TRUNCATE cases CASCADE;" +
    "TRUNCATE gender_identities CASCADE;" +
    "TRUNCATE case_note_actions CASCADE;" +
    "TRUNCATE audits CASCADE;" +
    "TRUNCATE case_tags CASCADE;" +
    "TRUNCATE tags CASCADE;" +
    "TRUNCATE civilian_titles CASCADE;" +
    "TRUNCATE districts CASCADE;" +
    "TRUNCATE case_classifications CASCADE;" +
    "TRUNCATE notifications CASCADE;";

  await models.sequelize.query(truncationQuery, {
    type: models.sequelize.QueryTypes.RAW
  });
};
