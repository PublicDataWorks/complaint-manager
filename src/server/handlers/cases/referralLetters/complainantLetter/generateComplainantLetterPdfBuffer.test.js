import { cleanupDatabase } from "../../../../testHelpers/requestTestHelpers";
import { generateReferralLetterBodyAndAuditDetails } from "../generateReferralLetterBodyAndAuditDetails";
import timekeeper from "timekeeper";
import Case from "../../../../../sharedTestHelpers/case";
import { CASE_STATUS } from "../../../../../sharedUtilities/constants";
import models from "../../../../policeDataManager/models";
import Civilian from "../../../../../sharedTestHelpers/civilian";
import generateComplainantLetterPdfBuffer, {
  generateComplainantLetterHtml
} from "./generateComplainantLetterPdfBuffer";

let existingCase, timeOfDownload, complainant;

const AWS = require("aws-sdk");
jest.mock("aws-sdk");

jest.mock("html-pdf", () => ({
  create: (html, pdfOptions) => ({
    toBuffer: callback => {
      callback(null, html);
    }
  })
}));

jest.mock("../generateReferralLetterBodyAndAuditDetails");

afterEach(async () => {
  await cleanupDatabase();
  generateReferralLetterBodyAndAuditDetails.mockReset();
  timekeeper.reset();
});

afterAll(async () => {
  await models.sequelize.close();
});

beforeEach(async () => {
  timeOfDownload = new Date("2018-07-01 19:00:22 CDT");
  timekeeper.freeze(timeOfDownload);

  let s3 = AWS.S3.mockImplementation(() => ({
    config: {
      loadFromPath: jest.fn(),
      update: jest.fn()
    }
  }));

  complainant = new Civilian.Builder()
    .defaultCivilian()
    .withId(undefined)
    .withCivilianTitle({ name: "Miss", id: 2 })
    .build();

  const caseAttributes = new Case.Builder()
    .defaultCase()
    .withId(12070)
    .withFirstContactDate("2017-12-25")
    .withIncidentDate("2016-01-01")
    .withComplainantCivilians([complainant]);

  existingCase = await models.cases.create(caseAttributes, {
    include: [
      {
        model: models.civilian,
        as: "complainantCivilians",
        auditUser: "someone"
      }
    ],
    auditUser: "test"
  });

  await existingCase.update(
    { status: CASE_STATUS.ACTIVE },
    { auditUser: "test" }
  );
  await existingCase.update(
    { status: CASE_STATUS.LETTER_IN_PROGRESS },
    { auditUser: "test" }
  );
});

describe("generateComplainantLetterPdfBuffer", function () {
  test("generates complainant letter pdf html correctly", async () => {
    const letterHtml = await generateComplainantLetterHtml(
      existingCase,
      complainant
    );

    expect(letterHtml).toMatchSnapshot();
  });

  test("pdf buffer is created for complainant letter", async () => {
    const pdfResults = await generateComplainantLetterPdfBuffer(
      existingCase,
      complainant
    );

    expect(pdfResults).toMatchSnapshot();
  });

  // test("pdf buffer should generate complainant letter with case reference prefiC when primary complainant is anonymized", async () => {
  //   const civilian = await models.civilian.findOne({
  //     where: { caseId: existingCase.id }
  //   });
  //
  //   await models.civilian.update(
  //     {
  //       isAnonymous: true
  //     },
  //     {
  //       where: {
  //         id: civilian.id
  //       },
  //       auditUser: "test user"
  //     }
  //   );
  //   const newCase = await models.cases.findOne({
  //     where: { id: existingCase.id },
  //     include: [
  //       {
  //         model: models.civilian,
  //         as: "complainantCivilians",
  //         auditUser: "someone"
  //       }
  //     ],
  //     auditUser: "someone"
  //   });
  //
  //   const pdfResults = await generateComplainantLetterPdfBuffer(
  //     newCase,
  //     newCase.primaryComplainant
  //   );
  //   expect(pdfResults).toMatchSnapshot();
  // });
});
