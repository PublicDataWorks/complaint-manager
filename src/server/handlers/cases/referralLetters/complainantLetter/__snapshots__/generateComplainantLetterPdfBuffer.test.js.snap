// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateComplainantLetterPdfBuffer generates complainant letter pdf html correctly 1`] = `
"<html>
    <head>
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\"/>
    </head>
    <body>
        <div id=\\"pageHeader-first\\">
            <div style=\\"text-align: center;\\">
                <img style=\\"max-width: 223px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/header_text.png\\" />
            </div>
            <div style=\\"text-align: center; margin-top: 8px\\">
                <img style=\\"max-width: 42px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/icon.ico\\" />
            </div>
        </div>
        <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
            Chuck<br/>
            July 01, 2018<br/>
            Page {{page}}
        </div>

        <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/icon.ico\\" />
            </span>
            <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
            <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
        </div>
        <p style=\\"color: #7F7F7F;\\">
            SUSAN HUTSON
            <br/>
            INDEPENDENT POLICE MONITOR
        </p>
        <p><br/></p>
        <p>
        July 01, 2018
        </p>
        <p><br/></p>
        <p>
        Miss Berry
            <p>123 Main St, Fl 2, Mayo St & Mustard Ave, Sandwich, IL 63456 (In front of Wal-Mart)</p>
            <p>cberry@cberry.com</p>
        </p>
        <p><br/></p>
        <p>Re: OIPM Complaint# CC2017-0001</p>
        <p><br/></p>
        <p>Dear 
             Miss Berry:
        </p>
        <p><br/></p>
        <p>
            On December 25, 2017, you contacted the Office of the Independent Police Monitor
            (OIPM) alleging possible misconduct by an officer of the New Orleans Police
            Department (NOPD or Department) for possible violations of several NOPD rules. As a
            result of your contact with us, OIPM No. CC2017-0001 was generated.
        </p>
        <p><br/></p>
        <p>
            Among other things, the OIPM takes complaints and examines the NOPD’s internal investigations system by conducting independent reviews of completed investigations into allegations of misconduct to determine whether they have been conducted appropriately. The OIPM does not conduct separate or new investigations.
        </p>

        <p><br/></p>
        <p>We forwarded the information you provided us to the NOPD’s Public Integrity Bureau
            (PIB) as an inquiry and asked that the matter be reviewed for possible violations of the
            NOPD rules and regulations. You may be contacted by a representative of PIB or by an
            NOPD supervisor regarding this matter.</p>
        <p><br/></p>
        <p>Please take into consideration that facts and/or allegations from your complaint may be
            used in future OIPM reports. If facts and/or allegations are used in future OIPM
            reports, names of witnesses, law enforcement, and complainants may be included. Once
            the review of your investigation has been completed, all or parts of the complaint may
            become public records.</p>
        <p><br/></p>
        <p>Enclosed you will find a copy of the OIPM letter to PIB and some information about
            our office. If you have any questions regarding the status of the information you
            provided, please contact us at (504) 309-9799 or via email at
            policemonitor@nolaipm.gov. Please refer to your OIPM No. CC2017-0001 when you
            contact our office.</p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p>
        Abe Lowe IV,
            <p><br/></p>
            <p><br/></p>
        Complaint Intake Specialist
        </p>
    </body>
</html>
"
`;

exports[`generateComplainantLetterPdfBuffer pdf buffer is created for complainant letter 1`] = `
"<html>
    <head>
        <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"pdf.css\\"/>
    </head>
    <body>
        <div id=\\"pageHeader-first\\">
            <div style=\\"text-align: center;\\">
                <img style=\\"max-width: 223px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/header_text.png\\" />
            </div>
            <div style=\\"text-align: center; margin-top: 8px\\">
                <img style=\\"max-width: 42px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/icon.ico\\" />
            </div>
        </div>
        <div id=\\"pageHeader\\" style=\\"font-size:8.5pt; color: #7F7F7F;\\">
            Chuck<br/>
            July 01, 2018<br/>
            Page {{page}}
        </div>

        <div id=\\"pageFooter\\" style=\\"text-align: center; margin-top: 16px\\">
            <span  style=\\"display:inline-block; margin: 6px 16px 0 0\\">
                <img style=\\"max-width: 30px\\" src=\\"file:/app/src/server/handlers/cases/referralLetters/assets/icon.ico\\" />
            </span>
            <span style=\\"display:inline-block; font-size:7pt; color: #7F7F7F;\\">
                INDEPENDENT POLICE MONITOR <br />
                2714 Canal Street, Suite 201 | NEW ORLEANS, LOUISIANA | 70119 <br />
                Phone (504) 309-9799| Fax (504) 309-7345
            </span>
            <span  style=\\"display:inline-block; width: 46px\\">&nbsp;</span>
        </div>
        <p style=\\"color: #7F7F7F;\\">
            SUSAN HUTSON
            <br/>
            INDEPENDENT POLICE MONITOR
        </p>
        <p><br/></p>
        <p>
        July 01, 2018
        </p>
        <p><br/></p>
        <p>
        Miss Berry
            <p>123 Main St, Fl 2, Mayo St & Mustard Ave, Sandwich, IL 63456 (In front of Wal-Mart)</p>
            <p>cberry@cberry.com</p>
        </p>
        <p><br/></p>
        <p>Re: OIPM Complaint# CC2017-0001</p>
        <p><br/></p>
        <p>Dear 
             Miss Berry:
        </p>
        <p><br/></p>
        <p>
            On December 25, 2017, you contacted the Office of the Independent Police Monitor
            (OIPM) alleging possible misconduct by an officer of the New Orleans Police
            Department (NOPD or Department) for possible violations of several NOPD rules. As a
            result of your contact with us, OIPM No. CC2017-0001 was generated.
        </p>
        <p><br/></p>
        <p>
            Among other things, the OIPM takes complaints and examines the NOPD’s internal investigations system by conducting independent reviews of completed investigations into allegations of misconduct to determine whether they have been conducted appropriately. The OIPM does not conduct separate or new investigations.
        </p>

        <p><br/></p>
        <p>We forwarded the information you provided us to the NOPD’s Public Integrity Bureau
            (PIB) as an inquiry and asked that the matter be reviewed for possible violations of the
            NOPD rules and regulations. You may be contacted by a representative of PIB or by an
            NOPD supervisor regarding this matter.</p>
        <p><br/></p>
        <p>Please take into consideration that facts and/or allegations from your complaint may be
            used in future OIPM reports. If facts and/or allegations are used in future OIPM
            reports, names of witnesses, law enforcement, and complainants may be included. Once
            the review of your investigation has been completed, all or parts of the complaint may
            become public records.</p>
        <p><br/></p>
        <p>Enclosed you will find a copy of the OIPM letter to PIB and some information about
            our office. If you have any questions regarding the status of the information you
            provided, please contact us at (504) 309-9799 or via email at
            policemonitor@nolaipm.gov. Please refer to your OIPM No. CC2017-0001 when you
            contact our office.</p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p><br/></p>
        <p>
        Abe Lowe IV,
            <p><br/></p>
            <p><br/></p>
        Complaint Intake Specialist
        </p>
    </body>
</html>
"
`;
